### PR TITLE
dma: HDA rename prefix from cAVS to ADSP

### PR DESCRIPTION
**Pull request by teburd**
HDA is a common IP used across the entire ADSP line and
deserves a name respecting that alongside similiar IP drivers such as
the ADSP GPDMA driver.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

### CODE REVIEW

File: drivers/dma/Kconfig.intel_adsp_hda [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Intel ADSP HDA configuration options
+
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_INTEL_INTEL_ADSP_HDA_HOST_IN := intel,adsp-hda-host-in
+DT_COMPAT_INTEL_INTEL_ADSP_HDA_HOST_OUT := intel,adsp-hda-host-out
+DT_COMPAT_INTEL_INTEL_ADSP_HDA_LINK_IN := intel,adsp-hda-link-in
+DT_COMPAT_INTEL_INTEL_ADSP_HDA_LINK_OUT := intel,adsp-hda-link-out
```
Comment (MaureenHelm) [inline drivers/dma/Kconfig.intel_adsp_hda:9]: INTEL_INTEL?

File: drivers/dma/dma_intel_adsp_hda.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_DMA_DMA_INTEL_ADSP_HDA_COMMON_H_
+#define ZEPHYR_DRIVERS_DMA_DMA_INTEL_ADSP_HDA_COMMON_H_
```
Comment (MaureenHelm) [inline drivers/dma/dma_intel_adsp_hda.h:8]: DMA_DMA?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Some funny search and replace errors, but I like the direction

File: drivers/dma/Kconfig.intel_adsp_hda (hunk 1 above)
Comment (teburd) [reply drivers/dma/Kconfig.intel_adsp_hda:9]: dohhhhh thanks, find/replace fail

File: drivers/dma/dma_intel_adsp_hda.h (hunk 2 above)
Comment (teburd) [reply drivers/dma/dma_intel_adsp_hda.h:8]: that was there before dma/dma_intel_adsp_hda_common.h

File: drivers/dma/Kconfig.intel_adsp_hda (hunk 1 above)
Comment (nashif) [reply drivers/dma/Kconfig.intel_adsp_hda]: developer developer

**Review by MaureenHelm — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]

**Pull request merged**